docs/reference/snapshot-restore/apis/get-snapshot-api.asciidoc

```diff
@@ -323,3 +333,137 @@ The API returns the following response:
 // TESTRESPONSE[s/"end_time": "2020-07-06T21:55:18.129Z"/"end_time": $body.responses.0.snapshots.0.end_time/]
 // TESTRESPONSE[s/"end_time_in_millis": 1593094752018/"end_time_in_millis": $body.responses.0.snapshots.0.end_time_in_millis/]
 // TESTRESPONSE[s/"duration_in_millis": 0/"duration_in_millis": $body.responses.0.snapshots.0.duration_in_millis/]
+
+The following request returns information for all snapshots with prefix `snapshot` in the `my_repository` repository,
+limiting the response size to 2 and sorting by snapshot name.
+
+[source,console]
+----
+GET /_snapshot/my_repository/snapshot*?size=2&sort=name
+----
+
+The API returns the following response:
+
+[source,console-result]
+----
+{
+  "responses": [
+    {
+      "repository": "my_repository",
+      "snapshots": [
+        {
+          "snapshot": "snapshot_1",
+          "uuid": "dKb54xw67gvdRctLCxSket",
+          "version_id": <version_id>,
+          "version": <version>,
+          "indices": [],
+          "data_streams": [],
+          "feature_states": [],
+          "include_global_state": true,
+          "state": "SUCCESS",
+          "start_time": "2020-07-06T21:55:18.129Z",
+          "start_time_in_millis": 1593093628850,
+          "end_time": "2020-07-06T21:55:18.129Z",
+          "end_time_in_millis": 1593094752018,
+          "duration_in_millis": 0,
+          "failures": [],
+          "shards": {
+            "total": 0,
+            "failed": 0,
+            "successful": 0
+          }
+        },
+        {
+          "snapshot": "snapshot_2",
+          "uuid": "vdRctLCxSketdKb54xw67g",
+          "version_id": <version_id>,
+          "version": <version>,
+          "indices": [],
+          "data_streams": [],
+          "feature_states": [],
+          "include_global_state": true,
+          "state": "SUCCESS",
+          "start_time": "2020-07-06T21:55:18.130Z",
+          "start_time_in_millis": 1593093628851,
+          "end_time": "2020-07-06T21:55:18.130Z",
+          "end_time_in_millis": 1593094752019,
+          "duration_in_millis": 1,
+          "failures": [],
+          "shards": {
+            "total": 0,
+            "failed": 0,
+            "successful": 0
+          }
+        }
+      ],
+      "next": "snapshot_2,snapshot_2"
```

Comment:
I wonder if we should just not allow pagination when doing a query across multiple repos. The current approach of doing pagination per repo becomes really confusing when multiple `next` values are returned but only one `after` can be set. Maybe this doesn't matter too much in the short-run as it's an 8.x only issue anyway but for `8 we should have something nicer?

Reply:
Hmm that does seem suboptimal, I think it would be useful to have pagination even if multiple repos are involved. Why do we need to return multiple `next` values? Since we're using the same sort order for all of them can we not use the minimum of the `next` values? Relatedly, does this mean that the `limit` applies to each repository separately rather than across all of them?

Reply:
> Relatedly, does this mean that the limit applies to each repository separately rather than across all of them?

Yes, currently that's how it's implemented (mainly because of the focus on getting this out the door for 7.x where this is a non-concern because we only work with a single repo at a time and I didn't want to create different logic for `master` and `7.x` for now to keep the backport reasonably simple). 


> Since we're using the same sort order for all of them can we not use the minimum of the next values? 
> Why do we need to return multiple next values?

We can implement pagination globally across all repos but then the logic for master will diverge a lot from `7.x` which I'd rather not do right now.  If we don't do that though, I don't see an alternative to either forbidding pagination with multiple repos in the query or having multiple next values?

Reply:
Ok, forbidding it seems the most future-proof answer today, it's easy to become more permissive in future but much harder to change something that works into something else that works differently.

I think it'd be better for the `next` field to be at the top level of the response tho rather than having it inside the (single) snapshot object.

Reply:
The problems/confusion here has been resolved last week in https://github.com/elastic/elasticsearch/pull/74451. We now have just a single `next` field at the top level and snapshots are paginated across repositories :)